torch/csrc/Storage.h

```diff
@@ -5,17 +5,114 @@
 
 #define THPStorageStr "torch.UntypedStorage"
 
+namespace c10 {
+
+template <>
+struct MaybeOwnedTraits<c10::Storage> {
```

Comment:
NOTE: I imagine these OwnedTraits defs should be moved to `c10/core/Storage.h`, but I have them here for now to reduce build time if I need to modify them